### PR TITLE
Fixed FAB crashing issue

### PR DESCRIPTION
**Pull request by distantfutures**
navigating to contactsSelectFragment caused crash while in Calls tab. Fixed by removing optionmenu inflation from VP fragments and letting HomeFragment handle optionmenu inflation. removed uncessary code accordingly

### CODE REVIEW

**Review by OskarLasota — COMMENTED** [review]
hey, not sure which change it was but when i go from chats tab -> calls tab -> press fab -> press back -> chats tab ( navigation to chats tab doesnt work )

File: WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/contactsselect/ContactsSelectFragment.kt [hunk 1]
```diff
@@ -143,20 +143,30 @@ class ContactsSelectFragment : BaseFragment<FragmentContactsSelectBinding>(), Ac
...
-        super.onPrepareOptionsMenu(menu)
-        menu.findItem(R.id.searchFragment).isVisible = false
-        menu.findItem(R.id.itemLinkedDevices).isVisible = false
-        menu.findItem(R.id.itemStarredMessages).isVisible = false
-        menu.findItem(R.id.itemSettings).isVisible = false
-        menu.findItem(R.id.itemNewGroup).isVisible = false
-        menu.findItem(R.id.itemNewBroadcast).isVisible = false
-    }
-
+//    override fun onPrepareOptionsMenu(menu: Menu) {
```
Comment (NelzkieLabs) [inline WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/contactsselect/ContactsSelectFragment.kt]: I think this should be remove if not need anymore. 😁

File: WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/home/HomeFragment.kt [hunk 2]
```diff
@@ -138,6 +142,21 @@ class HomeFragment : BaseFragment<FragmentHomeBinding>() {
         }
     }
 
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        Log.i(TAG, "TAB POSITION $currentTabPosition")
```
Comment (NelzkieLabs) [inline WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/home/HomeFragment.kt]: Maybe can remove debug log?

File: WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/home/calls/CallsFragment.kt [hunk 3]
```diff
@@ -115,23 +116,21 @@ class CallsFragment : BaseFragment<FragmentCallsBinding>(), ActionMode.Callback,
...
-    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
-        inflater.inflate(R.menu.home_calls_menu, menu)
-        super.onCreateOptionsMenu(menu, inflater)
-    }
-
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         return when(item.itemId) {
             R.id.search -> {
                 viewModel.searchClick()
+                Log.i("CALLTEST", "Search Clicked!")
```
Comment (NelzkieLabs) [inline WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/home/calls/CallsFragment.kt]: also here. do we need the debug log?

**Comment by distantfutures** [comment]
> hey, not sure which change it was but when i go from chats tab -> calls tab -> press fab -> press back -> chats tab ( navigation to chats tab doesnt work )

Just tried to work through it, haven't found a solution yet. same thing seems to happen when: status tab -> press fab -> press back -> chats tab (navigation also doesn't work)

In either scenarios above, after you popped off the fab fragment (ContactsSelect or Camera), press the chats tab, pressed the fab again, then back again, it retains the previous tab you were at before (ContactsSelect or Camera)

Thought it was a lifecycle issue with remembering the position after navigating to a different fragment(non-tab fragment) in homeFragment so I tried storing the position in the homeViewModel but that didnt work either. 
Looked at creating global actions to navigate to the other fragments, didn't work either. 
Seems like a ViewPager issue as Josh had said but MAY be closer related to the HomeStatePagerAdapter.

**Review by OskarLasota — APPROVED** [review]

**Pull request merged**